src/compiler/checker.ts

```diff
@@ -5396,15 +5435,179 @@ namespace ts {
+         */
+        function hasLateBoundName(node: Declaration): node is LateBoundDeclaration {
+            const name = getNameOfDeclaration(node);
+            return name && isLateBoundName(name);
+        }
+
+        /**
+         * Tests whether a declaration name is definitely late-bindable.
+         */
+        function isLateBoundName(node: DeclarationName): node is LateBoundName {
+            return isComputedPropertyName(node)
+                && isEntityNameExpression(node.expression)
+                && isLateBoundNameType(checkComputedPropertyName(node));
+        }
+
+        function isLateBoundNameType(type: Type): type is LiteralType | UniqueESSymbolType {
+            return !!(type.flags & TypeFlags.StringOrNumberLiteralOrUnique);
+        }
+
+        /**
+         * Performs late-binding of the dynamic members of a declaration.
+         */
+        function lateBindMembers(lateMembers: UnderscoreEscapedMap<TransientSymbol>, node: Declaration) {
+            const members = getMembersOfDeclaration(node);
+            if (members) {
+                for (const member of members) {
+                    if (hasLateBoundName(member)) {
+                        lateBindMember(lateMembers, node.symbol, member);
+                    }
+                }
+            }
+        }
+
+        /**
+         * Tests whether a declaration has a dynamic name that cannot be late-bound.
+         */
+        function hasUnboundDynamicName(node: Declaration) {
+            return hasDynamicName(node) && !hasLateBoundName(node);
+        }
+
+        /**
+         * Tests whether a declaration name is a dynamic name that cannot be late-bound.
+         */
+        function isUnboundDynamicName(node: DeclarationName) {
+            return isDynamicName(node) && !isLateBoundName(node);
+        }
+
+        /**
+         * Gets the symbolic name for a late-bound member from its type.
+         */
+        function getLateBoundNameFromType(type: LiteralType | UniqueESSymbolType) {
+            if (type.flags & TypeFlags.UniqueESSymbol) {
+                return `__@${type.symbol.escapedName}@${getSymbolId(type.symbol)}` as __String;
+            }
+            if (type.flags & TypeFlags.StringOrNumberLiteral) {
+                return escapeLeadingUnderscores("" + (<LiteralType>type).value);
+            }
+        }
+
+        /**
+         * Gets the late-bound symbol for a symbol (if it has one).
+         */
+        function getLateBoundSymbol(symbol: Symbol) {
+            if (symbol && (symbol.flags & SymbolFlags.ClassMember) && symbol.escapedName === InternalSymbolName.Computed) {
+                const links = getSymbolLinks(symbol);
+                if (!links.lateSymbol) {
+                    links.lateSymbol = symbol;
+                    const parent = symbol.parent;
+                    if (parent && (parent.flags & SymbolFlags.HasMembers)) {
+                        const lateMembers = getLateBoundMembersOfSymbol(parent);

```

Comment:
@ahejlsberg, in an offline review you indicated you'd rather we just determined the late-bound symbol of a declaration without caring about the entire set of late-bound members of the container. The reason I need to reference the container's late-bound members is to properly handle merging accessor declarations, e.g.:

```ts
const x = Symbol();
const obj = {
  get [x]() { ... },
  set [x](value) { ... }
};
```